src/platform/ESP32/ESP32BasicPacketFilters.h

```diff
@@ -0,0 +1,85 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <inet/EndpointQueueFilter.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <inet/IPPacketInfo.h>
+#include <inet/InetInterface.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/Span.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace {
+
+using namespace chip;
+using namespace chip::Inet;
+
+class ESP32PacketFilter : public chip::Inet::EndpointQueueFilter
+{
+public:
+    /**
+     * @brief Initialize the packet filter with a starting limit
+     *
+     * @param maxAllowedQueuedPackets - max number of pending-in-queue not yet processed predicate-matching packets
+     */
+    ESP32PacketFilter() {}
+
+    void Init()
+    {
+        /* Packets which are causing Packet Buffer Pool to get empty
+         *  Destination IP fe80::0000:1111:2222:3333 indicates packets shared locally on device to device communication, not needed
```

Comment:
Can you point to documentation where it states that `fe80::0000:1111:2222:3333` is local? the address seems very human-crafted to me rather than some standard, so I find it odd.

ff02::fb is the address in https://www.rfc-editor.org/rfc/rfc6762.html  ... why do we filter that?

Reply:
Link-local addresses MUST NOT be filtered outright. Better rules must be used.